### PR TITLE
[chore] Release docs clarification

### PR DESCRIPTION
**Pull request by djaglowski**
Based on a my experience releasing v0.83.0, I believe the following clarifications would help reduce potential errors in future releases.

### CODE REVIEW

File: .github/workflows/prepare-release.yml [hunk 1]
```diff
@@ -6,11 +6,11 @@ on:
     # the minor version number and set the patch number to 0.
     inputs:
       candidate-stable:
-        description: Release candidate version (stable, like 1.0.0-rc4). Don't include `v`.
+        description: Release candidate version (stable, like 1.0.0-rcv0014). Don't include `v`.
```
Comment (Aneurysm9) [inline .github/workflows/prepare-release.yml:9]:
```suggestion
        description: Release candidate version (stable, like 1.0.0-rcv0014). Don't include a leading `v`.
```

File: .github/workflows/prepare-release.yml [hunk 2]
```diff
@@ -6,11 +6,11 @@ on:
...
 
       current-stable:
         required: true
-        description: Current version (stable, like 1.0.0-rc3). Don't include `v`.
+        description: Current version (stable, like 1.0.0-rcv0014). Don't include `v`.
```
Comment (Aneurysm9) [inline .github/workflows/prepare-release.yml:13]:
```suggestion
        description: Current version (stable, like 1.0.0-rcv0014). Don't include a leading `v`.
```

File: docs/release.md [hunk 3]
```diff
@@ -28,12 +28,15 @@ It is possible that a core approver isn't a contrib approver. In that case, the
...
    Check if stable modules have any changes since the last release by running `make check-changes PREVIOUS_VERSION=v1.0.0-rc9 MODSET=stable`. If there are no changes, there is no need to release new version for stable modules. 
 
-2. Manually run the action [Automation - Prepare Release](https://github.com/open-telemetry/opentelemetry-collector/actions/workflows/prepare-release.yml). When prompted, enter the version numbers determined in Step 1, but do not include `v`. This action will create an issue to track the progress of the release and a pull request to update the changelog and version numbers in the repo. **While this PR is open all merging in Core should be haulted**. **Do not** move forward until this PR is merged.
+2. Manually run the action [Automation - Prepare Release](https://github.com/open-telemetry/opentelemetry-collector/actions/workflows/prepare-release.yml). This action will create an issue to track the progress of the release and a pull request to update the changelog and version numbers in the repo. **While this PR is open all merging in Core should be haulted**.
+   - When prompted, enter the version numbers determined in Step 1, but do not include `v`.
```
Comment (Aneurysm9) [inline docs/release.md:32]:
```suggestion
   - When prompted, enter the version numbers determined in Step 1, but do not include a leading `v`.
```

File: docs/release.md [hunk 4]
```diff
@@ -45,14 +48,21 @@ It is possible that a core approver isn't a contrib approver. In that case, the
...
+   - Manually update `cmd/otelcontribcol/builder-config.yaml`
+   - Manually update `cmd/oteltestbedcol/builder-config.yaml`
+   - Run `make genotelcontribcol`
+   - Push updates to the PR
+   -  🛑 **Do not move forward until this PR is merged.** 🛑
 
 ## Releasing opentelemetry-collector-contrib
 
-1. Manually run the action [Automation - Prepare Release](https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/workflows/prepare-release.yml). When prompted, enter the version numbers determined in Step 1, but do not include `v`. This action will a pull request to update the changelog and version numbers in the repo. **While this PR is open all merging in Contrib should be haulted**. Do not move forward until this PR is merged.
+1. Manually run the action [Automation - Prepare Release](https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/workflows/prepare-release.yml). When prompted, enter the version numbers determined in Step 1, but do not include `v`. This action will a pull request to update the changelog and version numbers in the repo. **While this PR is open all merging in Contrib should be haulted**.
```
Comment (Aneurysm9) [inline docs/release.md:61]:
```suggestion
1. Manually run the action [Automation - Prepare Release](https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/workflows/prepare-release.yml). When prompted, enter the version numbers determined in Step 1, but do not include a leading `v`. This action will a pull request to update the changelog and version numbers in the repo. **While this PR is open all merging in Contrib should be haulted**.
```

**Review by codeboten — APPROVED** [review]
Thanks for the improvements 👍🏻

**Pull request merged**